lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -1082,7 +1086,7 @@ SILInstruction *SILCombiner::visitReleaseValueInst(ReleaseValueInst *RVI) {
 
     // retain_value of an enum_inst where we know that it has a payload can be
     // reduced to a retain_value on the payload.
-    if (EI->hasOperand()) {
+    if (EI->hasOperand() && !hasValueDeinit(EI)) {
```

Comment:
This `hasValueDeinit` is redundant because it's already checked in line 1078

Reply:
That redundant check is a result of multiple fixes that were merged into this PR. I'll cleanup the extra check on main first to avoid merge conflicts.

Reply:
That cleanup is now in https://github.com/apple/swift/pull/66314